packages/babel-generator/src/generators/types.ts

```diff
@@ -202,7 +202,7 @@ export function NumericLiteral(this: Printer, node: t.NumericLiteral) {
 export function StringLiteral(this: Printer, node: t.StringLiteral) {
   const raw = this.getPossibleRaw(node);
   if (!this.format.minified && raw != null) {
-    this.token(raw);
+    this.token(raw as string);
```

Comment:
Doesn't the `!=` null check above help TS with this? Maybe if we make the return type of `getPossibleRaw` `string | undefined` instead of `string | void` it works?